widgets/common/src/filters/RuleOperatorDropdown.tsx

```diff
@@ -1,28 +1,52 @@
 import { camelCase } from 'lodash';
-import type { FunctionComponent } from 'react';
+import { FunctionComponent, useEffect, useState } from 'react';
 import type { DropdownItemProps } from 'semantic-ui-react/dist/commonjs/modules/Dropdown/DropdownItem';
+
 import { i18nPrefix } from './consts';
+import { FilterRuleRowType, LabelsFilterRuleOperators, AttributesFilterRuleOperators } from './types';
+import type { FilterRuleOperator } from './types';
+
+interface RuleOperatorDropdownProps {
+    ruleType: FilterRuleRowType;
+    onChange: (value: FilterRuleOperator) => void;
+    value: FilterRuleOperator;
+}
+
+function getDropdownOptions(operators: string[]) {
+    const { i18n } = Stage;
+    return operators.map(
+        (operator): DropdownItemProps => ({
+            text: i18n.t(`${i18nPrefix}.operatorsLabels.${camelCase(operator as string)}`),
+            value: operator
+        })
+    );
+}
 
-const { i18n } = Stage;
-const operatorsOptions = [
-    'any_of',
-    'not_any_of',
-    'is_null',
-    'is_not_null',
-    'contain',
-    'not_contain',
-    'start_with',
-    'end_with'
-].map(
-    (operator): DropdownItemProps => ({
-        text: i18n.t(`${i18nPrefix}.operatorsLabels.${camelCase(operator)}`),
-        value: operator
-    })
-);
-
-const RuleOperatorDropdown: FunctionComponent = () => {
+const RuleOperatorDropdown: FunctionComponent<RuleOperatorDropdownProps> = ({ ruleType, onChange, value }) => {
     const { Dropdown } = Stage.Basic;
 
-    return <Dropdown search selection name="ruleOperator" options={operatorsOptions} />;
+    const [options, setOptions] = useState([] as DropdownItemProps[]);
+
+    useEffect(() => {
+        const operators = Object.values(
+            ruleType === FilterRuleRowType.Label ? LabelsFilterRuleOperators : AttributesFilterRuleOperators
+        );
+        const dropdownOptions = getDropdownOptions(operators);
+
+        setOptions(dropdownOptions);
+    }, [ruleType]);
```

Comment:
This looks like a poor man's `useMemo` :smile: 

```suggestion
    const options = useMemo(() => {
        const operators = Object.values(
            ruleType === FilterRuleRowType.Label ? LabelsFilterRuleOperators : AttributesFilterRuleOperators
        );
        return getDropdownOptions(operators);
    }, [ruleType]);
```

Reply:
Good idea, I'll change it.